sandbox/grist/formula_prompt.py

```diff
@@ -107,21 +110,50 @@ def visible_columns(engine, table_id):
   ]
 
 
-def class_schema(engine, table_id, exclude_col_id=None):
+def class_schema(engine, table_id, exclude_col_id=None, lookups=False):
   result = "@dataclass\nclass {}:\n".format(table_id)
+
+  if lookups:
+    tmp = []
+    tmp2 = []
+    for col_id, col in visible_columns(engine, table_id):
+      if col_id != exclude_col_id:
+        tmp.append(col_id + ' = None')
+        tmp2.append(f'{col_id}={col_id}')
+    tmp.append('sort_by = None')
+    tmp2.append('sort_by=sort_by')
+    args = ', '.join(tmp)
+    args2 = ', '.join(tmp2)
+    result += f"     def __len__(self):\n"
```

Comment:
Grist tables don't support this, why have it here?

Reply:
Otherwise OpenAI starts to feel our model is ORMy and expects a count() method that doesn't exist. It leads to results that work. I played fast and loose with type of lookupRecords. This is just playing around in a Draft PR.

Reply:
But doesn't it suggest `len(Table1)` instead of `Table1.count()`?

Reply:
Sure, but it is there to suggest `len(Table1.lookupRecords(...))` instead of `Table1.lookupRecords(...).count()` which doesn't work today and probably couldn't be added because it would collide with `count` column names.

Listen, don't read too much into this. I can push to `paulfitz/grist-core` if it bothers you for me to play in this branch. There's no plan currently to land this PR.

Reply:
> Sure, but it is there to suggest len(Table1.lookupRecords(...)) instead of Table1.lookupRecords(...).count()

Ah, so it's about counting filtered subsets. I was wondering why it needed to count the whole table.

I'm suprised it was trying `.count`, did it do that even with the `-> List[{table_id}]` annotation?

> if it bothers you for me to play in this branch.

It doesn't bother me, I'm just interested in the conversation and want to be in the loop.

> There's no plan currently to land this PR.

But is there a plan to allow/suggest people to try out this branch themselves?

Reply:
> But is there a plan to allow/suggest people to try out this branch themselves?

Yes, we'll point to this PR and a docker image for anyone who wants to get hands-on.